internals/overlord/checkstate/handlers.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package checkstate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"gopkg.in/tomb.v2"
+
+	"github.com/canonical/pebble/internals/logger"
+	"github.com/canonical/pebble/internals/overlord/state"
+	"github.com/canonical/pebble/internals/plan"
+)
+
```

Comment:
The design feels like it will work nicely for me.

Based on some examples I've studied and how I try to think about changes / tasks (which could well be wrong), there are a couple "bonus features"  you can pull out of the hat if you do it in a specific way, which may or may not be useful for every case I am sure:

- If multiple tasks use the same data, you can ```Set```/```Get``` the data at the change level
- Since the changes / tasks can act as an audit trial, I think it becomes super interesting to have the data associated with the request ```Set```, so that success and failure contexts are clear for something that happened 1 month ago, and for which the actual details could have been updated in the plan, without changing the check name, for example.
- In the latter case, it means the change does not directly need to monitor changes in its parameters, but relies on top level management to Abort it when it is no longer applicable.

I wonder ... just brainstorming here for ideas, which again may be unwanted for the case at hand ... 
```
Check { HTTPTasks { PerformHTTPCheck ; RecoverHTTPCheck } }
Check { TCPTasks { PerformTCPCheck ; RecoverTCPCheck } }
Check { ExecTasks { PerformExecCheck ; RecoverExecCheck } }
```

One other way I would imagine one can spin this is to say that one change is tracking a check for its lifetime, and the perform and recover task is type specific, so that the change can have check ```Set``` data that describes what has to be done. Top level management would simply Abort when the change is out of date, and crate a new change new updated or new checks.

Reply:
I've reworked this quite a bit now. Commentary in main thread.